apm-sniffer/apm-sdk-plugin/elastic-job-3.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/elasticjob/ElasticJobExecutorInterceptor.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.apm.plugin.elasticjob;
+
+import org.apache.shardingsphere.elasticjob.api.listener.ShardingContexts;
+import org.apache.skywalking.apm.agent.core.context.ContextCarrier;
+import org.apache.skywalking.apm.agent.core.context.ContextManager;
+import org.apache.skywalking.apm.agent.core.context.trace.AbstractSpan;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.EnhancedInstance;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.InstanceMethodsAroundInterceptor;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.MethodInterceptResult;
+import org.apache.skywalking.apm.network.trace.component.ComponentsDefine;
+
+import java.lang.reflect.Method;
+
+public class ElasticJobExecutorInterceptor implements InstanceMethodsAroundInterceptor {
+    @Override
+    public void beforeMethod(EnhancedInstance objInst, Method method, Object[] allArguments, Class<?>[] argumentsTypes,
+                             MethodInterceptResult result) throws Throwable {
+        ShardingContexts shardingContexts = (ShardingContexts) allArguments[0];
+        Integer item = (Integer) allArguments[1];
+        String operateName = ComponentsDefine.ELASTIC_JOB.getName() + "/" + shardingContexts.getJobName();
+        AbstractSpan span = ContextManager.createEntrySpan(operateName, new ContextCarrier());
+        span.setComponent(ComponentsDefine.ELASTIC_JOB);
+        span.tag("item", item == null ? "" : String.valueOf(item));
+        span.tag("shardingTotalCount", Integer.toString(shardingContexts.getShardingTotalCount()));
+        span.tag("taskId", shardingContexts.getTaskId());
+        span.tag("shardingItemParameters", shardingContexts.getShardingItemParameters() == null ? "" : shardingContexts.getShardingItemParameters().toString());
```

Comment:
@zhaoyuguang Tag key should be added as a constant in the plugin.

Reply:
@tristaZero This is how we get these tags.

Reply:
Hi @wu-sheng It seems these tags come from [ShardingContexts](https://github.com/apache/shardingsphere-elasticjob/blob/master/elasticjob-api/src/main/java/org/apache/shardingsphere/elasticjob/api/listener/ShardingContexts.java), which is a necessary job context for users. I am not familiar with what content skyWalking needs for its plugin, but if it wants to get core job information, I think it is enough. :)

Reply:
BTW, there are other props from `ShardingContexts`, like `jobName`, `jobParameter`, and so forth. I am not sure whether you @zhaoyuguang  @wu-sheng need them or not. Thanks.